.devcontainer/devcontainer.json

```
@@ -0,0 +1,24 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the
+// README at: https://github.com/devcontainers/templates/tree/main/src/python
+{
+	"name": "Python 3",
+	// Or use a Dockerfile or Docker Compose file. More info: https://containers.dev/guide/dockerfile
+	"image": "mcr.microsoft.com/devcontainers/universal:2",
+	// Features to add to the dev container. More info: https://containers.dev/features.
+	// "features": {},
+	// Use 'forwardPorts' to make a list of ports inside the container available locally.
+	"forwardPorts": [
+		5003
+	],
+	// Use 'postCreateCommand' to run commands after the container is created.
+	"postCreateCommand": "pip install -r requirements.txt",
+	"postAttachCommand": "chmod +x ${containerWorkspaceFolder}/.devcontainer/addcodespacename.sh; ${containerWorkspaceFolder}/.devcontainer/addcodespacename.sh; python main.py",
```

Checking in the `addcodespace.sh` script with the appropriate linux file permissions should be enough here (doing the `chmod` ahead of time).

---

```suggestion
	"postAttachCommand": ".devcontainer/addcodespacename.sh && python main.py",
```

Lifecycle hooks are executed [from the workspaceFolder](https://containers.dev/implementors/json_reference/#lifecycle-scripts), so this command can be simplified as shown.  Additionally, i'd suggest using the `&&` and operator, so that if the left side fails, the entire command fails without continuing on.

---

Hmm when I remove the chmod, it says permission denied. How do I do the chmod ahead of time? I'm a little confused on this suggestion.

---

I'm putting it back for now and then when I see your suggestion I'll update it!

---

nvm, silly mistake I made. Thank you!